app/controllers/user_controller.py

```diff
@@ -1,18 +1,72 @@
+from http import HTTPStatus
+
+from app.configs.database import db
+from app.models.address_model import AddressModel
+from app.models.city_model import CityModel
+from app.models.user_model import UserModel
+from flask import jsonify, request
+from sqlalchemy.orm import Session
+from werkzeug.exceptions import NotFound
+
+
 def signup():
-    ...
+    session: Session = db.session
+    data = request.get_json()
+
+    city = data.pop("city")
+    cep = data.pop("cep")
+
+    try:
+
+        city_query = session.query(CityModel).filter_by(name=city).first_or_404()
+        cep_query = session.query(AddressModel).filter_by(cep=cep).first()
+
+        new_user = UserModel(**data)
+
+        if cep_query:
+            new_user.address = cep_query
+        else:
+            new_cep = AddressModel(cep=cep)
+            new_cep.city = city_query
+            new_user.address = new_cep
+
+        session.commit()
+    except NotFound:
+        cities = session.query(CityModel).all()
+
+        return {
```

Comment:
I think it would be nice to create an exception folder and create a custom one for this case, so it don't take much space in the request function, there is no need to do that now but we still have a lot of logic to implement here and if we don't take care, this function will be huge.

Reply:
I think it's best to do a refactoring as soon as the application starts to grow.